File: game_card/Game card/utils/deck.py
import random

class Card:
    def __init__(self, suit, rank, value):
        self.suit = suit
        self.rank = rank
        self.value = value
        # Les couleurs rouges pour l'interface
        self.color = "red" if suit in ['♥', '♦'] else "black"

    def __str__(self):
        return f"{self.rank}{self.suit}"

class Deck:
    def __init__(self):
        self.cards = []
        self.build()

    def build(self):
        suits = ['♥', '♦', '♣', '♠']
        # Valeurs par défaut (adaptables selon le jeu)
        ranks = [('2', 2), ('3', 3), ('4', 4), ('5', 5), ('6', 6), ('7', 7), 
                 ('8', 8), ('9', 9), ('10', 10), ('V', 10), ('D', 10), ('R', 10), ('A', 11)]
        self.cards = [Card(s, r, v) for s in suits for r, v in ranks]
        random.shuffle(self.cards)

    def draw(self):
        return self.cards.pop() if self.cards else None

File: game_card2/game_card2/lib/models/deck.dart
import 'dart:math';

import 'card_model.dart';

class Deck {
  final List<CardModel> cards = [];

  Deck() {
    build();
  }

  void build() {
    cards.clear();
    const suits = ['♥', '♦', '♣', '♠'];
    final ranks = <Map<String, Object>>[
      {'rank': '2', 'value': 2},
      {'rank': '3', 'value': 3},
      {'rank': '4', 'value': 4},
      {'rank': '5', 'value': 5},
      {'rank': '6', 'value': 6},
      {'rank': '7', 'value': 7},
      {'rank': '8', 'value': 8},
      {'rank': '9', 'value': 9},
      {'rank': '10', 'value': 10},
      {'rank': 'V', 'value': 10},
      {'rank': 'D', 'value': 10},
      {'rank': 'R', 'value': 10},
      {'rank': 'A', 'value': 11},
    ];

    for (final suit in suits) {
      for (final rankMap in ranks) {
        cards.add(CardModel(
          suit: suit,
          rank: rankMap['rank'] as String,
          value: rankMap['value'] as int,
        ));
      }
    }

    shuffle();
  }

  void shuffle() {
    cards.shuffle(Random());
  }

  CardModel? draw() {
    if (cards.isEmpty) return null;
    return cards.removeLast();
  }
}
